### DISCUSSION TITLE
Documentation Subdomain for Simpler URLS

### DISCUSSION THREAD

**Discussion by Mordil** [SwiftPackageIndex/SwiftPackageIndex-Server#2431] [Ideas]
Right now documentation is hosted as a subpage of a _version_, so if you want to type in directly you first need to know the version name, even if I just want the 'default' loaded docs (#2331).

This can be painful for discovery and quick access when I don't know all these details

## Example

If I was to type in the URL directly, I'd have to write

`https://swiftpackageindex.com/mordil/redistack/2.0.0-gamma.1/documentation/redistack/`

vs

`https://docs.swiftpackageindex.com/mordil/redistack` which could auto-redirect to the expanded default `../<version>/<module>`

## Future Direction
By also having a `docs` sub-domain it also precludes being able to have a separate index page for exploring _all_ available documentation hosted by SPI.

**Comment by Mordil** [comment]
The argument could be made that the solution would be to deprecate the `<owner>/<repo>/<version>/documentation/<module>/` URL format for a new `.../documentation/<version>/<module>` format instead of a whole separate sub-domain and redirect logic

**Reply by Mordil** [reply]
Correct me if I'm wrong, the reason ya'll did it this way was to avoid the question of "what do I see if I do `/documentation` just by itself?

**Reply by finestructure** [reply]
If you tack on `/documentation` to a package page we redirect you to the latest documentation version:

https://swiftpackageindex.com/Mordil/RediStack/documentation


Does that help?

**Reply by Mordil** [reply]
Oh, I totally didn't know that. I don't know why I didn't even think to try.

My discovery process was -> new button & feature -> use it -> notice URL pattern -> think that's the only way to access docs

**Reply by daveverwer** [reply]
Yea we've spent quite a bit of time trying to figure out the best way to communicate this. Until a couple of weeks ago the "Documentation" button on the package page had this redirect as its target so that if anyone copied the URL from there, they'd get a link that worked together, but as part of attempting to fix Google's problems with indexing our documentation pages, that link now (either temporarily or permanently) now points direct to the final destination.

**Comment by daveverwer** [comment]
> separate index page for exploring all available documentation hosted by SPI.

This is a great idea, and something I'd love to explore more in the long term, but it wouldn't be on a new subdomain, and may not even be on a separate page. I'd love to add more rich content to the homepage, like highlighting newly created articles or package documentation updates.

**Reply by Mordil** [reply]
Ah, yeah that's totally doable.

I don't think a separate subdomain is truly necessary now that I know `/documentation` works just fine
